camera_control/camera_control/camera/realsense.py

```diff
@@ -0,0 +1,97 @@
+import pyrealsense2 as rs
+import numpy as np
+
+
+class RealSense:
+    def __init__(self, logger, rgb_size, depth_size):
+        self.logger = logger
+        self.rgb_size = rgb_size
+        self.depth_size = depth_size
+
+    def initialize(self):
+        # Setup code based on https://github.com/IntelRealSense/librealsense/blob/master/wrappers/python/examples/align-depth2color.py
+        self.pipeline = rs.pipeline()
+        self.config = rs.config()
+        self.config.enable_stream(
+            rs.stream.depth, self.depth_size[0], self.depth_size[1], rs.format.z16, 30
+        )
+        self.config.enable_stream(
+            rs.stream.color, self.rgb_size[0], self.rgb_size[1], rs.format.bgr8, 30
+        )
+        self.profile = self.pipeline.start(self.config)
+        self.setup_inten_extrins()
+
+        depth_sensor = self.profile.get_device().first_depth_sensor()
+        self.depth_scale = depth_sensor.get_depth_scale()
+
+        self.depthMinMeters = 0.1
+        self.depthMaxMeters = 10
+
+    def setup_inten_extrins(self):
+        color_prof = self.profile.get_stream(rs.stream.color)
+        depth_prof = self.profile.get_stream(rs.stream.depth)
+
+        self.depth_intrins = depth_prof.as_video_stream_profile().get_intrinsics()
+        self.color_intrins = color_prof.as_video_stream_profile().get_intrinsics()
+        self.logger.info(f"Color Frame intrinsics:{self.color_intrins}")
+        self.depth_to_color_extrinsic = depth_prof.get_extrinsics_to(color_prof)
+        self.color_to_depth_extrinsic = color_prof.get_extrinsics_to(depth_prof)
+
+    def get_clipping_distance(self):
+        # Getting the depth sensor's depth scale (see rs-align example for explanation)
+        depth_sensor = self.profile.get_device().first_depth_sensor()
+        depth_scale = depth_sensor.get_depth_scale()
+
+        # We will be removing the background of objects more than
+        #  clipping_distance_in_meters meters away
+        clipping_distance_in_meters = 1  # 1 meter
+        self.clipping_distance = clipping_distance_in_meters / depth_scale
+
+    def get_frames(self):
+        frames = self.pipeline.poll_for_frames()
```

Comment:
nit: I think we should use wait_for_frames so that we're guaranteed that a frame is available (which it will be given that we'll always be running runner/laser detection at a lower rate than the camera capture FPS). There's minimal overhead to calling wait_for_frames

Reply:
When I first built out the project I used the wait_for_frames calls and ran into a number of issues. I believe that the call was causing the node to block, which caused other issues in communication etc. I think long term we might want to see if there is a way to add a callback to the realsense camera fires every time a frame is available without having to wait or poll within the camera node. We could also create a async call that re-calls get_frame on a shorter timescale then the period if we want to try and prevent frame loss.

Reply:
hmm interesting, it shouldn't block (should be equivalent to the poll call in terms of overhead) given a frame is available. No big deal to keep as is